source/tit/geom/partitioning/sfc_partitioning.hpp

```diff
@@ -0,0 +1,109 @@
+/* ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ *\
+ * Part of the Tit Solver project, under the MIT License.
+ * See /LICENSE.md for license information. SPDX-License-Identifier: MIT
+\* ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
+
+#pragma once
+
+#include <algorithm>
+#include <ranges>
+#include <vector>
+
+#include "tit/core/basic_types.hpp"
+#include "tit/core/profiler.hpp"
+#include "tit/core/type_traits.hpp"
+
+#include "tit/geom/sort.hpp"
+
+namespace tit::geom {
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Space filling curve based partitioning.
+template<sort_factory SortFactory,
```

Comment:
I am unsure if I like this being a factory or not. Maybe we should accept `perm` as a parameter, while sorting is to be produced in factory?